clarity_ext/service/clarity_service.py

```diff
@@ -82,3 +86,50 @@ def update_single(self, domain_object, ignore_commit):
                 self.clarity_repository.update(api_resource)
         else:
             raise NotImplementedError("The type '{}' isn't implemented".format(type(domain_object)))
+
+    def get_project_by_name(self, project_name):
+        project_resource = utils.single(self.session.api.get_projects(name=project_name))
+        return ProjectClarityMapper.create_object(project_resource)
+
+    def create_container(self, container, with_samples=False, assign_to=None):
+        """
+        Creates the container and all samples in it. 
+
+        Requires a container and samples that do not have an ID. The samples are interpreted as
+        original samples, not analytes.
+        """
+        if container.id:
+            raise AssertionError("This container already has an ID: {}".format(container))
+        container_type = utils.single(
+                self.session.api.get_containertypes(name=container.container_type))
+        
+        container_res = entities.Container.create(
+                self.session.api, name=container.name, type=container_type)
+        container.id = container_res.id
+
+        if not with_samples:
+            return container
+
+        created_artifacts = list()
+
+        # TODO: Do this in a batch call
+        for well in container.occupied:
+            sample = well.artifact
+            sample_res = entities.Sample.create(
+                    self.session.api,
+                    container=container_res,
+                    position=repr(well.position),
+                    name=sample.name,
+                    project=sample.project.api_resource,
+                    udfs=sample.udf_map.to_dict())
+
+            artifact = entities.Artifact(self.session.api, id=sample_res.id + "PA1")
+            created_artifacts.append(artifact)
+
+        if assign_to:
+            # Assign all the samples directly to a workflow
+            workflow = utils.single(self.session.api.get_workflows(name=assign_to))
+            self.session.api.route_artifacts(created_artifacts, workflow_uri=workflow.uri)
```

Comment:
Is this working without entering a step?

Reply:
Yes, it will be assigned to the first step in the workflow.